src/App.tsx

```diff
@@ -1,26 +1,68 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosSlice } from './features/todos';
+import { Todo } from './types/Todo';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState('');
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then((data: Todo[]) => dispatch(todosSlice.actions.setTodos(data)))
+      .finally(() => setIsLoading(false));
+  }, [dispatch]);
+
+  const preparedTodos = () => {
```

Comment:
this is not a good name for a function - preparedTodos sounds like it's already an array. But let's keep the name and change logic: wrap it with `useMemo` and add correct dependencies array as second argument